
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <title>Peak Search &#8212; Lanceotron  documentation</title>
    <link rel="stylesheet" href="_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script async="async" type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Welcome to Lanceotron’s documentation!" href="index.html" />
   
  <link rel="stylesheet" href="_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />

  </head><body>
  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="peak-search">
<h1>Peak Search<a class="headerlink" href="#peak-search" title="Permalink to this headline">¶</a></h1>
<div class="section" id="summary">
<h2>Summary<a class="headerlink" href="#summary" title="Permalink to this headline">¶</a></h2>
<p>Peak Search allows you to call peaks from a wig file and classify them. Initially peaks are only called on a single chromosome. From this the correct parameters and a model can be chosen in order to call peaks on the entire genome. The following steps are required:-</p>
<ul class="simple">
<li><strong>Parameter Sweep</strong> Peaks are called on a supplied Wig file out using a number of different parameters. A bed file is created for each combination of parameters.</li>
<li><strong>Initial Peak Calling</strong> The initial peak calling can be analysed by looking at how changing different parameters effects the distribution of the peaks. From this, the appropriate parameters are chosen and images and stats are generated for all peaks generated with the chosen parameters</li>
<li><strong>Tagging/Create Model</strong> From the peak images and stats, the peaks can be tagged (classified) by the user into different categories. These categories can then submitted to a machine learning step, which generates a model.</li>
<li><strong>Genome Wide Classification</strong> Peaks are called and classified on the whole genome using the default model. In addition, newly generated models used for genome wide classification.</li>
</ul>
</div>
<div class="section" id="parameter-sweep">
<h2>Parameter Sweep<a class="headerlink" href="#parameter-sweep" title="Permalink to this headline">¶</a></h2>
<img alt="alternate text" class="align-center" src="_images/parameter_sweep.png" />
<ol class="arabic simple">
<li>Type the name and description of the project into the inputs (1 and 2 above)</li>
<li>Enter the http address of a publically accessible  Wig file into the text box(3). Once you have pressed enter or lost focus, the path entered will be checked to see if it is indeed a wig file and then the chromosomes displayed in the right hand drop down (4) for you to choose from.</li>
<li>You can submit the job by pressing the appropriate button(5) An email will be sent once the initial peak calling has finished.. You can also return the page to check the status</li>
</ol>
<p>If you just want to choose different parameters from a parameter sweep that has previously been carried out, then press ‘Use Exisiting’ (6) and choose the appropiate sweep.</p>
</div>
<div class="section" id="initial-peak-calling">
<h2>Initial Peak Calling<a class="headerlink" href="#initial-peak-calling" title="Permalink to this headline">¶</a></h2>
<div class="section" id="choosing-parameters">
<h3>Choosing Parameters<a class="headerlink" href="#choosing-parameters" title="Permalink to this headline">¶</a></h3>
</div>
<div class="section" id="processing-peaks">
<h3>Processing Peaks<a class="headerlink" href="#processing-peaks" title="Permalink to this headline">¶</a></h3>
<img alt="alternate text" class="align-center" src="_images/initial_peak_calls.png" />
<p>Once the parameters have been chosen and submitted, the following steps will be carried out on all the peaks
identified on the selected chromosome:-</p>
<ul class="simple">
<li>Peak statistics (width, height, area) are calculated</li>
<li>Overlap with any genes, annotations are calculated</li>
<li>Thumbnail created for each peak</li>
<li>tSNE and PCA carried out bases on peak shape</li>
<li>Peaks are classified using the default model and assigned a Peak and Noise score</li>
<li>Images are uplaoded to Zegami and a collection is created</li>
</ul>
<p>You will be informed about which stage is currently running. You do not have to remain on the page, but can return to it later to see
if all the processing has been completed.</p>
<p>Also, whilst the above is being carried out, peaks for the whole genome (not just the specified chromosome)
will be called and classified according to the parameters initially chosen and the default model.</p>
<p>Once processing is complete, you will have access to the Tagging/Create Model and Whole Genome Classification tabs.</p>
</div>
</div>
<div class="section" id="taggingcreating-model">
<h2>TaggingCreating Model<a class="headerlink" href="#taggingcreating-model" title="Permalink to this headline">¶</a></h2>
<div class="section" id="tagging-using-mlv">
<h3>Tagging Using MLV<a class="headerlink" href="#tagging-using-mlv" title="Permalink to this headline">¶</a></h3>
<div class="section" id="general-navigation">
<h4>General Navigation<a class="headerlink" href="#general-navigation" title="Permalink to this headline">¶</a></h4>
<p>Press the ‘Tag With MLV’ button ob the Tagging/Create Model panel and you will be taken to the MLV page</p>
<img alt="alternate text" class="align-center" src="_images/mlv_tagging_1.png" />
<p>Initially all the peaks will be shown in the right-hand tab and the the total number on the far right (3)
The graphs on the left show peak statistics, clustering (tSNE/PCA) and also scores given to the peaks using the
default model. In addition, overlap with any annotations will also be shown in pie charts, if this was specified.</p>
<p>Peaks can be filtered by selecting areas on the graphs in the left hand panels. Areas in the scatter plots
can be selected by pressing shift and dragging the mouse. Normal dragging will cause panning and zooming
can be achieved using the mouse wheel. Filtered peaks are also reflected in the bottom browser window and
are coloured by Peak score from green (score=1) to red (score =0).Image size can be altered using the slider (6)</p>
</div>
<div class="section" id="tagging-images">
<h4>Tagging Images<a class="headerlink" href="#tagging-images" title="Permalink to this headline">¶</a></h4>
<img alt="alternate text" class="align-center" src="_images/mlv_tagging_2.png" />
<p>Pressing the tag icon (4) will open the tagging dialog, now clicking on an image will then tag that image with the tag selected in the dialog,
or remove the tag, if none was selected. Multiple images can be tagged at once,  by pressing shift and clicking, which
tags all the images between the last and currently tagged item. Closing the tagging dialog will remove coloured border from the image, but not remove
the tags, and the images will become highlighted again once the dialog is reopened.</p>
<p>Once enough tags have been selected the tags need to be fixed with the ‘Fix Tags’ button (4). This opens a dialog which shows the current set, which initially will
be original. If enough images have been tagged you can the proceed by pressing the ‘Fix’ button . This stores all the tags in a database and allows a model to
be created from them.</p>
</div>
</div>
<div class="section" id="tagging-using-zegami">
<h3>Tagging Using Zegami<a class="headerlink" href="#tagging-using-zegami" title="Permalink to this headline">¶</a></h3>
<img alt="alternate text" class="align-center" src="_images/zegami_tagging_1.png" />
<p>Click the Zegami icon to go to the Zegami set (1),  you need to login <a class="reference external" href="https://app.zegami.com/accounts/login/">here</a> using
the correct password/username, otherwise a request error will be shown on the page. Then tag the sets in Zegami,
see the <a class="reference external" href="https://zegami.com/">Zegami web site</a>  for instructions on how to tag images.
When you have finished tagging, you will need to return to the main project page to proceed. Every time the project page is loaded it will
retrieve the current tags from Zegami, but this can also be done manually, with the ‘Refresh Tags’ button (2). Once you are happy with the tags, press the
‘fix tags’ button (3). Once the tags are fixed you can create a model see <a class="reference internal" href="#classifying-generating-a-model">Classifying/Generating a Model</a></p>
</div>
<div class="section" id="classifying-generating-a-model">
<h3>Classifying/Generating a Model<a class="headerlink" href="#classifying-generating-a-model" title="Permalink to this headline">¶</a></h3>
<img alt="alternate text" class="align-center" src="_images/creating_model_1.png" />
<p>Pressing the ‘Create Model’ button (1) will bring up a dialog, showing the number and type of peaks which have been tagged (2)
The source of these tags is also shown. In this example, only the original set is present (5) hence all tags are taken from this.
If other sets are present, for each peak, the tag will be taken from the last created set, if there is not one present, then
the next set will be examined for tag and so on until the original set is reached.</p>
<p>An appropriate name should be given to the model (3). The base model can also be changed from default to a previously created one (4).
Finally, the model can be submitted(6)
At the top of the screen, a message will show that the model is being created and a dialog will appear to inform you when the job is
finished. Once the model has been created you can view it either in Zegami or MLV. The model can then be used to call peaks genome
wide.</p>
<div class="section" id="viewing-in-mlv">
<h4>Viewing in MLV<a class="headerlink" href="#viewing-in-mlv" title="Permalink to this headline">¶</a></h4>
<img alt="alternate text" class="align-center" src="_images/creating_model_2.png" />
<p>Select the model in model dropdown (1) and the Noise,Peak Score graphs will update to reflect the new model</p>
</div>
<div class="section" id="viewing-in-zegami">
<h4>Viewing in Zegami<a class="headerlink" href="#viewing-in-zegami" title="Permalink to this headline">¶</a></h4>
<img alt="alternate text" class="align-center" src="_images/creating_model_3.png" />
<p>For each model created An extra field will be added to the original zegami collection with the title model_name tag score
(see above)</p>
</div>
</div>
<div class="section" id="working-with-sets">
<h3>Working with Sets<a class="headerlink" href="#working-with-sets" title="Permalink to this headline">¶</a></h3>
<div class="section" id="in-mlv">
<h4>In MLV<a class="headerlink" href="#in-mlv" title="Permalink to this headline">¶</a></h4>
<img alt="alternate text" class="align-center" src="_images/set_management_1.png" />
<p>The set dialog can be opened by clicking on the sets label (1). Once tags have been fixed, pressing the plus sign
next to a tag (2) will create a new set. This may take a while as peaks are re-clustered. Once this is complete
the view will automatically change to the reflect the set, with the charts showing the new
clustering (3). You can than tag this set and fix the tags in the normal way - see <a class="reference internal" href="#tagging-using-mlv">Tagging Using MLV</a>
To switch between sets press the eye icon and the graphs and images will update accordingly. As the number of setsexpands,
you can zoom (mouse wheel) and pan (mouse drag)in the diagram.</p>
</div>
<div class="section" id="in-zegami">
<h4>In Zegami<a class="headerlink" href="#in-zegami" title="Permalink to this headline">¶</a></h4>
<p>Sets can be created in the Tagging/Create Model tab of the main project page. Once tags have been fixed in a set,
pressing the plus icon next to a tag will create a set from those tags (see above). Once the set has been
created, it can be viewed in Zegami (using the zegami icon in the set), although you may have to wait a while for the set to be processed.</p>
<img alt="alternate text" class="align-center" src="_images/set_management_2.png" />
<p>The new zegami collection will consist only of items in the new set and contain extra fields for PCA/tSNE1, which
are prefixed with the set name.The set can be tagged and fixed in the normal way <a class="reference internal" href="#tagging-using-zegami">Tagging Using Zegami</a></p>
<p>For each set created, all the fields relevent to the sets (tSNE/PCA, tags) will also be added to the original Zegami collection
(prefixed with the set name) allowing comparison between sets</p>
</div>
</div>
</div>
<div class="section" id="genome-wide-classification">
<h2>Genome Wide Classification<a class="headerlink" href="#genome-wide-classification" title="Permalink to this headline">¶</a></h2>
<img alt="alternate text" class="align-center" src="_images/gw_classification_1.png" />
<p>Peaks in the whole genome are tagged and classified according to the model chosen. Classification automatically
runs for the default model. Other models can also be rub and these can be selected by using the ‘change’ button and
then selecting the appropriate model from the table.</p>
<p>All classifications, either running or complete are shown in the table. Once complete the Average scores
of each tag submiited will be displayed, as well as icon which allows downloading of a bigbed file containing all the peak calls
and associated scores.</p>
<p>You can aslo view the results of the classification in a viewer by presssing the eye icon.</p>
</div>
</div>


          </div>
          
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<h1 class="logo"><a href="index.html">Lanceotron</a></h1>








<h3>Navigation</h3>
<p class="caption"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Peak Search</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#summary">Summary</a></li>
<li class="toctree-l2"><a class="reference internal" href="#parameter-sweep">Parameter Sweep</a></li>
<li class="toctree-l2"><a class="reference internal" href="#initial-peak-calling">Initial Peak Calling</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#choosing-parameters">Choosing Parameters</a></li>
<li class="toctree-l3"><a class="reference internal" href="#processing-peaks">Processing Peaks</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#taggingcreating-model">TaggingCreating Model</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#tagging-using-mlv">Tagging Using MLV</a></li>
<li class="toctree-l3"><a class="reference internal" href="#tagging-using-zegami">Tagging Using Zegami</a></li>
<li class="toctree-l3"><a class="reference internal" href="#classifying-generating-a-model">Classifying/Generating a Model</a></li>
<li class="toctree-l3"><a class="reference internal" href="#working-with-sets">Working with Sets</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#genome-wide-classification">Genome Wide Classification</a></li>
</ul>
</li>
</ul>

<div class="relations">
<h3>Related Topics</h3>
<ul>
  <li><a href="index.html">Documentation overview</a><ul>
      <li>Previous: <a href="index.html" title="previous chapter">Welcome to Lanceotron’s documentation!</a></li>
  </ul></li>
</ul>
</div>
<div id="searchbox" style="display: none" role="search">
  <h3>Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" />
      <input type="submit" value="Go" />
      <input type="hidden" name="check_keywords" value="yes" />
      <input type="hidden" name="area" value="default" />
    </form>
    </div>
</div>
<script type="text/javascript">$('#searchbox').show(0);</script>








        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="footer">
      &copy;2018, Martin Sergeant.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 1.8.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="_sources/peak_search.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    
  </body>
</html>